typing/typecore.ml

```diff
@@ -3543,9 +3564,33 @@ and type_cases ?in_function env ty_arg ty_res ?conts partial_flag loc caselist =
     match conts with
     | None -> List.map (fun (pat, env) -> (pat, env, None)) pat_env_list
     | Some conts ->
-        List.map2
-          (fun (pat, env) cont -> (pat, env, cont))
-          pat_env_list conts
+        let ty = newvar () in
+        (* remember original level *)
+        (* Create a fake abstract type declaration for effect type. *)
+        let level = get_current_level () in
+        let decl = {
+          type_params = [];
+          type_arity = 0;
+          type_kind = Type_abstract;
+          type_private = Public;
+          type_manifest = None;
+          type_variance = [];
+          type_newtype_level = Some (level, level);
+          type_loc = loc;
+          type_attributes = [];
+        }
+        in
+        Ident.set_current_time ty.level;
+        let name = Ctype.get_new_abstract_name "effect" in
+        let (id, new_env) = Env.enter_type name decl env in
+        Ctype.init_def(Ident.current_time());
+        let ty_eff = newgenty (Tconstr (Path.Pident id,[],ref Mnil)) in
+        let ty_cont = Predef.type_continuation ty_eff ty_res in
+        let pat_env_cont_listt = List.map2
+            (fun (pat, env) cont -> let (envv, _) = env in
+              (pat, env, type_continuation_pat envv ty_cont cont))
```

Comment:
@lpw25 How do I construct `ty_cont` that's passed to  here? I just tried to copy how it's done in `type_effect_cases`.

This is the current error message I am getting

→ OCAMLRUNPARAM=b  ./ocaml-multicore/boot/ocamlrun ./ocaml-multicore/ocamlc -nostdlib -I ./ocaml-multicore/stdlib -dtypedtree p.ml
File "p.ml", line 9, characters 16-18:
Error: This expression has type int but an expression was expected of type
         effect#1